common/list.c

```diff
@@ -298,3 +297,70 @@ list_dump_items(struct list *self)
         LOG_DEVEL(LOG_LEVEL_TRACE, "%d: %p", index, (void *) list_get_item(self, index));
     }
 }
+
+/******************************************************************************/
+/**
+ * Appends a string fragment to a list
+ * @param[in,out] start Pointer to start of fragment (by reference)
+ * @param end Pointer to one past end of fragment
+ * @param list List to append to
+ * @result 1 for success
+ *
+ * In the event of a memory failure, 0 is returned and the list is deleted.
+ */
+int
+append_fragment(const char **start, const char *end, struct list *list)
```

Comment:
Great idea to make this non-static so we can test it, as you have below. In this case however it's still a private function, it might be worth changing the name to something a bit less generic, like `split_string_append_fragment()`. What do you think?